contracts/test/Summa.ts

```diff
@@ -61,6 +48,129 @@ describe("Summa Contract", () => {
     };
   }
 
+  describe("deployment tests", () => {
+    it("should not deploy with invalid currencies", async () => {
+      const verifyingKey = await ethers.deployContract(
+        "src/VerifyingKey.sol:Halo2VerifyingKey"
+      );
+      await verifyingKey.deployed();
+
+      const snarkVerifier = await ethers.deployContract(
+        "src/SnarkVerifier.sol:Verifier"
+      );
+      await snarkVerifier.deployed();
+
+      await expect(
+        ethers.deployContract("Summa", [
+          verifyingKey.address,
+          snarkVerifier.address,
+          ["", "BTC"],
+          ["ETH", "BTC"],
+          8,
+        ])
+      ).to.be.revertedWith("Invalid cryptocurrency");
+
+      await expect(
+        ethers.deployContract("Summa", [
+          verifyingKey.address,
+          snarkVerifier.address,
+          ["ETH", "BTC"],
+          ["ETH", ""],
+          8,
+        ])
+      ).to.be.revertedWith("Invalid cryptocurrency");
+
+      await expect(
+        ethers.deployContract("Summa", [
+          verifyingKey.address,
+          snarkVerifier.address,
+          [],
+          ["ETH", ""],
+          8,
+        ])
+      ).to.be.revertedWith("Cryptocurrency names and chains number mismatch");
+    });
+
+    it("should not deploy with invalid byte range", async () => {
+      const verifyingKey = await ethers.deployContract(
+        "src/VerifyingKey.sol:Halo2VerifyingKey"
+      );
+      await verifyingKey.deployed();
+
+      const snarkVerifier = await ethers.deployContract(
+        "src/SnarkVerifier.sol:Verifier"
+      );
+      await snarkVerifier.deployed();
+
+      await expect(
+        ethers.deployContract("Summa", [
+          verifyingKey.address,
+          snarkVerifier.address,
+          ["ETH", "BTC"],
+          ["ETH", "BTC"],
+          0, // Invalid byte range
+        ])
+      ).to.be.revertedWith("Invalid balance byte range");
+    });
+
+    it("should not deploy with invalid snark verifier key", async () => {
+      const verifyingKey = await ethers.deployContract(
+        "src/VerifyingKey.sol:Halo2VerifyingKey"
+      );
+      await verifyingKey.deployed();
+      await expect(
+        ethers.deployContract("Summa", [
+          verifyingKey.address,
+          ethers.constants.AddressZero,
+          ["ETH", "BTC"],
+          ["ETH", "BTC"],
+          0, // Invalid byte range
```

Comment:
Regardless the revert message, should the 'balance byte range' parameter be explicitly set to 0 in other test cases?

Reply:
It shouldn't, thanks for pointing out!